plugins/ipam/dhcp/lease.go

```diff
@@ -344,6 +362,51 @@ func (l *DHCPLease) maintain() {
 	}
 }
 
+func (l *DHCPLease) checkLinkExistsWithBackoff(ctx context.Context) (bool, error) {
+	checkFunc := func() (bool, error) {
+		_, err := netlink.LinkByName(l.link.Attrs().Name)
+		if err != nil {
+			log.Printf("!bang: Error checking network link '%s': %v, type: %v", l.link.Attrs().Name, err, reflect.TypeOf(err))
+			var linkNotFoundErr *netlink.LinkNotFoundError
+			if errors.As(err, &linkNotFoundErr) {
+				log.Printf("Link not found error detected, no retry needed.")
+				return false, nil
+			}
+			// Log unexpected error types for debugging purposes
+			log.Printf("Unexpected error type: %v, retrying...", reflect.TypeOf(err))
+			return false, err
+		}
+		return true, nil
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, linkCheckTotalTimeout)
+	defer cancel()
+
+	exists, err := backoffRetryLinkExists(ctx, linkCheckRetryMax, checkFunc)
+	return exists, err
+}
+
+func backoffRetryLinkExists(ctx context.Context, maxDelay time.Duration, f func() (bool, error)) (bool, error) {
```

Comment:
I wound up creating another method because I needed to have the retry method ` f func() (bool, error)` return a bool and an error, so, I additionally added `linkCheckDelay0`